signature/verify.go

```diff
@@ -35,39 +36,64 @@ func verifyServiceRequest(v serviceRequest) error {
 	verificationHeader := v.GetVerificationHeader()
 	body := serviceMessageBody(v)
 	size := collection.Max(body.StableSize(), meta.StableSize(), verificationHeader.StableSize())
-	buf := make([]byte, 0, size)
-	return verifyServiceRequestRecursive(body, meta, verificationHeader, buf)
+	buffers := createBuffers(size)
+	return verifyServiceRequestRecursive(body, meta, verificationHeader, buffers)
 }
 
-func verifyServiceRequestRecursive(body stableMarshaler, meta *session.RequestMetaHeader, verify *session.RequestVerificationHeader, buf []byte) error {
+func createBuffers(size int) [3][]byte {
+	return [3][]byte{
+		make([]byte, 0, size),
+		make([]byte, 0, size),
+		make([]byte, 0, size),
```

Comment:
I am wondering if it makes sense to allocate once and use sub-slices here. Probably not, can bite us in an unexpected way.

Reply:
fixed